prober/grpc_health_v1.go

```diff
@@ -0,0 +1,50 @@
+package prober
+
+import (
+	"context"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/prometheus/blackbox_exporter/config"
+
+	"google.golang.org/grpc"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+)
+
+func ProbeGRPCHEALTHV1(ctx context.Context, target string, module config.Module, registry *prometheus.Registry, logger log.Logger) bool {
+	var (
+		conn    *grpc.ClientConn
+		parts   []string
+		service string
+		host    string
+	)
+
+	parts          = strings.Split(target, "/")
+	host, service  = parts[0], parts[1]
+	conn, err     := grpc.Dial(host, grpc.WithInsecure())
+
+	if err != nil {
+		level.Error(logger).Log("msg", "failed to connect")
+		return false
+	}
+
+	defer conn.Close()
+
+	req := &healthpb.HealthCheckRequest{Service: service}
+	client := healthpb.NewHealthClient(conn)
+	res, err := client.Check(ctx, req)
+
+	if err != nil {
+		level.Error(logger).Log("msg", err)
+		return false
+	}
+
+	if res.Status != 1 {
```

Comment:
Instead of `1`, this should be `healthpb.HealthCheckResponse_SERVING` (https://github.com/grpc/grpc-go/blob/7c1d326729dc9b0a07135f8902ddcc050ff8ab64/health/grpc_health_v1/health.pb.go#L31) - that makes the statement a bit more understandable (less magic) too.